be/src/column/column_builder.h

```diff
@@ -16,6 +18,7 @@ class ColumnBuilder {
     using DataColumnPtr = typename RunTimeColumnType<Type>::Ptr;
     using NullColumnPtr = NullColumn::Ptr;
     using DatumType = RunTimeCppType<Type>;
+    using MovableType = RunTimeCppMovableType<Type>;
```

Comment:
some DatumTypes are pointer types, objects referenced by pointers are owned by other pools, so arguments passed by right refs are invalid, so add static_assert over types to disallow misuse.

Reply:
For pointer type of column, their memory are allocated in the corresponding column, instead of outside mempool.
And they will be copied into the column for `Column::append(const T* obj)`, which also observe the above rule.

To avoid copy memory, this `MovableType` and corresponding `append(MovableType obj)` will move value into column.

For example:
```c++
JsonValue json;
builder.append(&json); // copy the json into builder
builder.append(std::move(json)); // move the json into builder without copy
```